packages/shared/friends/sagas.ts

```diff
@@ -1753,73 +1823,128 @@ export function getChannelInfo(request: GetChannelInfoPayload) {
 export function getChannelMembers(request: GetChannelMembersPayload) {
   const client: SocialAPI | null = getSocialClient(store.getState())
-  const fetchContentServer = getFetchContentUrlPrefix(store.getState())
   if (!client) return
 
   const channel = client.getChannel(request.channelId)
   if (!channel) return
 
-  const channelMembers: UpdateChannelMembersPayload = {
+  const channelMembersPayload: UpdateChannelMembersPayload = {
     channelId: request.channelId,
     members: []
   }
 
-  const channelMemberIds = channel.userIds?.slice(request.skip, request.skip + request.limit)
-  if (!channelMemberIds) {
+  const members = getMembers(client, channel.userIds ?? [], request.channelId)
+    .filter(([, name]) => {
+      const searchTerm = request.userName.toLocaleLowerCase()
+      const lowerCaseName = name.toLocaleLowerCase()
+      return lowerCaseName.search(searchTerm) >= 0
+    })
+    .slice(request.skip, request.skip + request.limit)
+
+  if (members.length === 0) {
     // it means the channel has no members
-    getUnityInstance().UpdateChannelMembers(channelMembers)
+    getUnityInstance().UpdateChannelMembers(channelMembersPayload)
     return
   }
 
-  // get the local part of the userId
-  const membersIds = channelMemberIds.map((id) => getUserIdFromMatrix(id))
+  const ownId = client.getUserId()
+
+  // update catalog with missing users, by using default profiles with name and image url
+  sendMissingProfiles(members, ownId)
 
-  // Todo Juli: use ensure friend
-  const filteredProfiles = getProfilesFromStore(store.getState(), membersIds, request.userName)
+  const userStatuses = client.getUserStatuses(...members.map(([id]) => id))
 
-  for (const profile of filteredProfiles) {
-    const member = membersIds.find((id) => id === profile.data.userId)
-    if (member) {
-      // Todo Juli: Check -- What we gonna do with isOnline?
-      channelMembers.members.push({ userId: member, isOnline: false })
+  for (const [memberId, memberName] of members) {
+    const userId = getUserIdFromMatrix(memberId)
+    if (ownId === memberId || userStatuses.get(memberId)?.presence === PresenceType.ONLINE) {
+      channelMembersPayload.members.push({
+        userId,
+        name: memberName,
+        isOnline: true
+      })
     }
   }
 
-  const profilesForRenderer = filteredProfiles.map((profile) =>
-    profileToRendererFormat(profile.data, {
-      baseUrl: fetchContentServer
-    })
-  )
-  getUnityInstance().AddUserProfilesToCatalog({ users: profilesForRenderer })
-  store.dispatch(addedProfilesToCatalog(filteredProfiles.map((profile) => profile.data)))
+  getUnityInstance().UpdateChannelMembers(channelMembersPayload)
+}
 
-  getUnityInstance().UpdateChannelMembers(channelMembers)
+/*
+ * Checks which members are present in the profile catalog and sends partial profiles for missing users
+ * @param members is an array of [member ID, name]
+ */
+function sendMissingProfiles(members: [string, string][], ownId: string) {
```

Comment:
Can this be more specific? for maintainability purposes, code is written once and read one thousand times

```suggestion
function sendMissingProfiles(members: {address: string, ownId: string}[], ownId: string) {
```